frontend/src/metabase/dashboard/components/DashboardGrid.tsx

```diff
@@ -221,48 +246,70 @@ class DashboardGridInner extends Component<
     }
   }
 
-  UNSAFE_componentWillReceiveProps(nextProps: DashboardGridProps) {
+  static getDerivedStateFromProps(
+    nextProps: DashboardGridProps,
+    state: DashboardGridState,
+  ) {
     const { dashboard, dashcardData, isEditing, selectedTabId } = nextProps;
+    const lastProps = state._lastProps;
+
+    if (
+      !lastProps ||
+      !_.isEqual(dashboard.dashcards, lastProps.dashboard.dashcards) ||
+      !_.isEqual(dashcardData, lastProps.dashcardData) ||
+      isEditing !== lastProps.isEditing ||
+      selectedTabId !== lastProps.selectedTabId
```

Comment:
Only trigger state updates if these props changed

Reply:
I'd rather you don't have this logic, I'm fine with it in https://github.com/metabase/metabase/pull/52885 because it existed before. This was never here, and the behavior would change. I'd rather keep the behavior closer to the before-refactor version.

Reply:
@WiNloSt good observation, since `getDerivedStateFromProps` only updates the keys you passed to it similar to `this.setState(...)` it should not override any existing states. Let me remove this check then

Reply:
fixed in https://github.com/metabase/metabase/pull/53213/commits/a7b1c5893ce46e3d2f6038a76363d48db1d0dfd2